File: Topic Based Problems/01 Fundamentals/1 Beginner/02 Basic IO/Java/BasicIO.java
import java.util.Scanner;

public class BasicIO {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter byteValue: ");
        byte byteValue = scanner.nextByte();

        System.out.print("Enter shortValue: ");
        short shortValue = scanner.nextShort();

        System.out.print("Enter intValue: ");
        int intValue = scanner.nextInt();

        System.out.print("Enter longValue: ");
        long longValue = scanner.nextLong();

        System.out.print("Enter longLongValue: ");
        long longLongValue = scanner.nextLong();

        System.out.print("Enter floatValue: ");
        float floatValue = scanner.nextFloat();

        System.out.print("Enter doubleValue: ");
        double doubleValue = scanner.nextDouble();

        System.out.print("Enter charValue: ");
        char charValue = scanner.next().charAt(0);

        System.out.print("Enter stringValue: ");
        String stringValue = scanner.next();

        System.out.println();

        System.out.println("byteValue: " + byteValue);
        System.out.println("shortValue: " + shortValue);
        System.out.println("intValue: " + intValue);
        System.out.println("longValue: " + longValue);
        System.out.println("longLongValue: " + longLongValue);
        System.out.println("floatValue: " + floatValue);
        System.out.println("doubleValue: " + doubleValue);
        System.out.println("charValue: " + charValue);
        System.out.println("stringValue: " + stringValue);

        scanner.close();
    }
}

File: Topic Based Problems/01 Fundamentals/1 Beginner/02 Basic IO/C/BasicIO.c
#include <stdio.h>

int main() {
    short shortValue;
    int intValue;
    long longValue;
    long long longLongValue;
    float floatValue;
    double doubleValue;
    long double longDoubleValue;
    char charValue;
    char stringValue[100]; // Assuming a maximum string length of 100

    printf("Enter shortValue: ");
    scanf("%hd", &shortValue);

    printf("Enter intValue: ");
    scanf("%d", &intValue);

    printf("Enter longValue: ");
    scanf("%ld", &longValue);

    printf("Enter longLongValue: ");
    scanf("%lld", &longLongValue);

    printf("Enter floatValue: ");
    scanf("%f", &floatValue);

    printf("Enter doubleValue: ");
    scanf("%lf", &doubleValue);

    printf("Enter longDoubleValue: ");
    scanf("%Lf", &longDoubleValue);

    printf("Enter charValue: ");
    scanf(" %c", &charValue); // Note the space before %c to consume the newline character

    printf("Enter stringValue: ");
    scanf(" %99s", stringValue); // Note the 99 to prevent buffer overflow

    printf("\n");

    printf("shortValue: %hd\n", shortValue);
    printf("intValue: %d\n", intValue);
    printf("longValue: %ld\n", longValue);
    printf("longLongValue: %lld\n", longLongValue);
    printf("floatValue: %f\n", floatValue);
    printf("doubleValue: %lf\n", doubleValue);
    printf("longDoubleValue: %Lf\n", longDoubleValue);
    printf("charValue: %c\n", charValue);
    printf("stringValue: %s\n", stringValue);
    
    return 0;
}
